### PR TITLE
[Form] Fixed regression: Empty values were not accepted anymore for collapsed, optional choice fields

### PR DESCRIPTION
**Pull request by webmozart**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14393 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php [hunk 1]
```diff
@@ -300,6 +303,27 @@ public function testSubmitSingleNonExpandedEmpty()
...
+        $this->assertTrue($form->isSynchronized());
+    }
+
+    public function testSubmitSingleNonExpandedEmptyExplicitEmptyChoice()
+    {
+        $form = $this->factory->create('choice', null, array(
+            'multiple' => false,
+            'expanded' => false,
+            'choices' => array(
+                'EMPTY_CHOICE' => 'Empty',
```
Comment (Tobion) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php]: doesn't this have to use choices_as_values: true to not use the deprecated functionality? I guess the trigger_error is missing for the old choices structure.

Comment (webmozart) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/ChoiceTypeTest.php]: You are right. I'll add the trigger_error statement in a separate PR.

**Comment by mbartok** [comment]
I still cannot submit optional entity filed (extended choice field) using empty_value with doctrine on Symfony 2.7 (e60f715). 

Field definition:

```php
$builder->add('dealer', 'entity', array(
    'label' => 'useraccount.label.dealer',
    'class' => 'MyAppBundle:Subject',
    'choice_label' => 'label',
    'empty_value' => '',
    'multiple' => false,
    'required' => false
));
```

After submitting empty value i get this error:

>SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for integer: ""

It seems like Symfony is trying to find Subject entity using empty string as id for dealer field:

>DBALException: An exception occurred while executing 'SELECT ... FROM subject s0_ WHERE s0_.id IN (?)' with params [""]:

**Comment by xabbuh** [comment]
see #14583

**Pull request merged**